src/cpp/rtps/builtin/discovery/endpoint/EDPSimple.cpp

```diff
@@ -675,7 +675,7 @@ void EDPSimple::assignRemoteEndpoints(const ParticipantProxyData& pdata)
     {
         logInfo(RTPS_EDP,"Adding SEDP Pub Writer to my Pub Reader");
         temp_writer_proxy_data_.guid().entityId = c_EntityId_SEDPPubWriter;
-        temp_writer_proxy_data_.persistence_guid().entityId = c_EntityId_SEDPPubWriter;
+        temp_writer_proxy_data_.persistence_guid().entityId = temp_writer_proxy_data_.persistence_guid().guidPrefix != c_GuidPrefix_Unknown ? c_EntityId_SEDPPubWriter : c_EntityId_Unknown;
```

Comment:
This comparison is repeated several times. We may add a set_persistence_entity_id method that performs the ternary operation of this line.

Reply:
Added a  RTPS_DllAPI void set_persistence_entity_id(const EntityId_t & nid) to WriterProxyData and replaced in the lines with the long ternary operation.